core/src/main/java/org/apache/calcite/avatica/remote/Driver.java

```diff
@@ -119,23 +119,25 @@ Service createService(AvaticaConnection connection, ConnectionConfig config) {
     final Service service;
     if (metaFactory != null) {
       service = metaFactory.create(connection);
-    } else if (config.url() != null) {
-      final AvaticaHttpClient httpClient = getHttpClient(connection, config);
-      final Serialization serializationType = getSerialization(config);
-
-      LOG.debug("Instantiating {} service", serializationType);
-      switch (serializationType) {
-      case JSON:
-        service = new RemoteService(httpClient);
-        break;
-      case PROTOBUF:
-        service = new RemoteProtobufService(httpClient, new ProtobufTranslationImpl());
-        break;
-      default:
-        throw new IllegalArgumentException("Unhandled serialization type: " + serializationType);
-      }
     } else {
-      service = new MockJsonService(Collections.<String, String>emptyMap());
+      if (config.url() != null) {
```

Comment:
This change seems to be eqivalent to the existing code but nests one more if.
While we want to make sure that we always use curly braces in control statements, else if a special case, which actually improves readability.

Reply:
Thanks @stoty.
While doing checkstyle formatting, I had accidentally done this. But restored the previous "if - else if -else" statement